### PR TITLE
Tests for IO Hooks

### PR DESCRIPTION
**Pull request by AlexHockey**
And here are the UTs for https://github.com/Metaswitch/cpp-common/pull/695

### CODE REVIEW

File: src/ut/utils_test.cpp [hunk 1]
```diff
@@ -198,3 +219,527 @@ TEST_F(UtilsTest, ParseIPAddresses)
   EXPECT_FALSE(parse_ip_target("1.2.3.4:8888", ai.address));
 }
 
+TEST_F(UtilsTest, Split)
```
Comment (eleanor-merry) [inline src/ut/utils_test.cpp:222]: What's the current coverage for utils now? Is it worth moving it out of coverage-not-yet and adding LCOVs for the places that still aren't covered.

File: src/ut/utils_test.cpp [hunk 2]
```diff
@@ -198,3 +219,527 @@ TEST_F(UtilsTest, ParseIPAddresses)
...
+{
```
Comment (eleanor-merry) [inline src/ut/utils_test.cpp:223]: Also, I assume the only changed/new tests are the stopwatch ones and the IO ones, yes?

File: src/ut/utils_test.cpp [hunk 3]
```diff
@@ -198,3 +219,527 @@ TEST_F(UtilsTest, ParseIPAddresses)
...
+
+  cwtest_advance_time_ms(11);
+  EXPECT_TRUE(_sw.stop());
+
+  unsigned long elapsed_us;
+  EXPECT_TRUE(_sw.read(elapsed_us));
+  EXPECT_EQ(ms_to_us(22), elapsed_us);
+}
+
+TEST_F(StopWatchTest, StopStartThenRead)
```
Comment (eleanor-merry) [inline src/ut/utils_test.cpp]: Isn't this test the same as the one above it?

**Review by eleanor-merry — APPROVED** [review]
LGTM

File: src/ut/utils_test.cpp (hunk 2 above)
Comment (AlexHockey) [reply src/ut/utils_test.cpp:223]: Correct.

File: src/ut/utils_test.cpp (hunk 3 above)
Comment (AlexHockey) [reply src/ut/utils_test.cpp]: Not quite. I the second test the stopwatch is running when the read happens. I have clarified by making the test name more descriptive and adding come more comments.

File: src/ut/utils_test.cpp (hunk 1 above)
Comment (AlexHockey) [reply src/ut/utils_test.cpp:222]: Still not very high. I'm going to leave it in coverage-not-yet. 

```
utils.cpp                                    325     236    72%   53,59,65-66,406-407,442,444,452,454,457-459,461-462,465-467,469,573,579,581,584,586-587,590,593-594,598-601,605,608,611-613,616,618,621,623,626-627,629,631,634,638,640,642,644,646,648,652,654,658,661-662,664,666,669,672,675,684-686,688,697,703,705,709,711-713,717,720,722-723,727,729,731,734-735,757,759,818,824-825,828,831
```

**Comment by AlexHockey** [comment]
@eleanor-merry I've made this PR point at overload-integration rather than master. I'm assigning to you to merge at a suitable time.

**Comment by AlexHockey** [comment]
In fact there is no overload-integration branch for this repo (fair enough) so I'll keep it pointed at master.

**Pull request merged**